core/domain/subtopic_validators_test.py

```diff
@@ -0,0 +1,857 @@
+
+        expected_output = [
+            (
+                u'[u\'failed validation check for '
+                'subtopic_page_commit_log_entry_ids field check of '
+                'SubtopicPageModel\', '
+                '[u"Entity id 0-1: based on field '
+                'subtopic_page_commit_log_entry_ids having value '
+                'subtopicpage-0-1-1, expected model '
+                'SubtopicPageCommitLogEntryModel '
+                'with id subtopicpage-0-1-1 but it doesn\'t exist"]]'),
+            u'[u\'fully-validated SubtopicPageModel\', 2]']
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+    def test_missing_snapshot_metadata_model_failure(self):
+        subtopic_models.SubtopicPageSnapshotMetadataModel.get_by_id(
+            '0-1-1').delete()
+        expected_output = [
+            (
+                u'[u\'failed validation check for snapshot_metadata_ids '
+                'field check of SubtopicPageModel\', '
+                '[u"Entity id 0-1: based on field snapshot_metadata_ids having '
+                'value 0-1-1, expected model SubtopicPageSnapshotMetadataModel '
+                'with id 0-1-1 but it doesn\'t exist"]]'),
+            u'[u\'fully-validated SubtopicPageModel\', 2]']
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+    def test_missing_snapshot_content_model_failure(self):
+        subtopic_models.SubtopicPageSnapshotContentModel.get_by_id(
+            '0-1-1').delete()
+        expected_output = [
+            (
+                u'[u\'failed validation check for snapshot_content_ids '
+                'field check of SubtopicPageModel\', '
+                '[u"Entity id 0-1: based on field snapshot_content_ids having '
+                'value 0-1-1, expected model SubtopicPageSnapshotContentModel '
+                'with id 0-1-1 but it doesn\'t exist"]]'),
+            u'[u\'fully-validated SubtopicPageModel\', 2]']
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+
+class SubtopicPageSnapshotMetadataModelValidatorTests(
+        test_utils.AuditJobsTestBase):
+
+    def setUp(self):
+        super(SubtopicPageSnapshotMetadataModelValidatorTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        self.signup(USER_EMAIL, USER_NAME)
+        self.user_id = self.get_user_id_from_email(USER_EMAIL)
+
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+
+        topics = [topic_domain.Topic.create_default_topic(
+            '%s' % i,
+            'topic%s' % i,
+            'abbrev-%s' % chr(120 + i),
+            'description%s' % i) for i in python_utils.RANGE(3)]
```

Comment:
It was this code which I left behind because it wasn't related to subtopic model 
The backend tests fail due to this?
rubrics = [
            skill_domain.Rubric(
                constants.SKILL_DIFFICULTIES[0], ['Explanation 1']),
            skill_domain.Rubric(
                constants.SKILL_DIFFICULTIES[1], ['Explanation 2']),
            skill_domain.Rubric(
                constants.SKILL_DIFFICULTIES[2], ['Explanation 3'])]
        skills = [
            skill_domain.Skill.create_default_skill(
                '%s' % i, 'skill%s' % i, rubrics)
            for i in python_utils.RANGE(9)]

        for skill in skills:
            skill_services.save_new_skill(self.owner_id, skill)

        stories = [story_domain.Story.create_default_story(
            '%s' % i,
            'title %d',
            'description %d' % i,
            '%s' % (python_utils.divide(i, 2)),
            'title-%s' % chr(97 + i)
        ) for i in python_utils.RANGE(6)]

        for story in stories:
            story_services.save_new_story(self.owner_id, story)

Reply:
This was removed as in one of the tests if was mentioned undefined

Reply:
@vojtechjelinek

Reply:
What I meant was moving whole classes that relate to subtopic models, if there is some part in the class that doesn't relate to subtopics it needs to be kept.

The task basically is to move four validation classes and four test classes for them, nothing else.

Reply:
@Rupeshkoushik Is this issue resolved now? You can leave a comment about it. Thanks!

Reply:
Yes,the issue is resolved thanks!